tests/src/test/scala/org/http4s/MessageSpec.scala

```diff
@@ -62,6 +62,34 @@ class MessageSpec extends Http4sSpec {
         Request(Method.GET).addCookie("token", "value").headers.get("Cookie".ci).map(_.value) must beSome("token=value")
       }
     }
+
+    "Request.copy" should {
+      val path1 = "/path1"
+      val path2 = "/somethingelse"
+
+      "update pathInfo after Request uri is changed, no caret" in {
+        val originalReq = Request(uri = Uri(path = path1))
+        val updatedReq = originalReq.copy(uri = Uri(path = path2))
+
+        originalReq.pathInfo mustEqual path1
+        updatedReq.pathInfo mustEqual path2
+      }
+
+      "update pathInfo after Request uri is changed, with caret" in {
+        val attributes = AttributeMap(Seq(AttributeEntry(Request.Keys.PathInfoCaret, 3)))
+
+        val originalReq = Request(uri = Uri(path = path1), attributes = attributes)
+        val updatedReq = originalReq.copy(uri = Uri(path = path2))
+
+        val caret = attributes.get(Request.Keys.PathInfoCaret).getOrElse(0)
+
+        originalReq.pathInfo mustNotEqual updatedReq.pathInfo
+        originalReq.pathInfo mustEqual originalReq.uri.path.splitAt(caret)._2
+        updatedReq.pathInfo mustEqual updatedReq.uri.path.splitAt(caret)._2
```

Comment:
This is testing that it still splits according to the caret, but says nothing about it resetting the caret.

I think the assertions here should be:

```scala
updatedReq.scriptName must_== ""
updatedReq.pathInfo must_== "/somethingelse"
```